code401challenges/src/main/java/tree/BinarySearchTree.java

```diff
@@ -42,6 +47,36 @@ public boolean contains(int data)
         return contains(root_Node, data);
     }
 
+    private ArrayList<Integer> breadthFirst(Node node, ArrayList<Integer> list)
```

Comment:
For this method, you don't have to take in a Node as an argument since we want to add the whole tree to the ArrayList... in other words we always want to start at the root.

Also, you put this method in your BST class which inherits from the Tree class. Therefore we can only use it on BST's. It would be better if you put it in the Tree class, then you could use this method on all Trees since nothing about the method would need to be changed.